cardano-config/src/Cardano/Config/Orphanage.hs

```diff
@@ -1,14 +1,85 @@
-{-# LANGUAGE StandaloneDeriving #-}
+{-# LANGUAGE FlexibleInstances #-}
 {-# LANGUAGE GeneralisedNewtypeDeriving #-}
+{-# LANGUAGE StandaloneDeriving #-}
+{-# LANGUAGE TypeSynonymInstances #-}
 {-# OPTIONS_GHC -Wno-orphans #-}
 
-module Cardano.Config.Orphanage ()
-  where
+module Cardano.Config.Orphanage
+  ( ConsensusTraceOptions
+  , ProtocolTraceOptions
+  , TraceOptions(..)
+  , traceOptionsDefault
+  ) where
 
-import Cardano.Prelude
+import           Cardano.Prelude
 
+import           Cardano.BM.Data.Tracer (TracingVerbosity (..))
 import qualified Ouroboros.Consensus.BlockchainTime as Consensus
-
+import qualified Ouroboros.Consensus.Node.Tracers as ConsensusTracers
+import           Ouroboros.Consensus.NodeNetwork (ProtocolTracers'(..))
 
 deriving instance Eq Consensus.SlotLength
 deriving instance Num Consensus.SlotLength
+
+type ConsensusTraceOptions = ConsensusTracers.Tracers' () ()    () (Const Bool)
+deriving instance Eq ConsensusTraceOptions
+deriving instance Show ConsensusTraceOptions
+
+type ProtocolTraceOptions  = ProtocolTracers'   () () ()    (Const Bool)
+deriving instance Eq ProtocolTraceOptions
+deriving instance Show ProtocolTraceOptions
+
+-- | Detailed tracing options. Each option enables a tracer
+--   which verbosity to the log output.
+data TraceOptions = TraceOptions
+  { traceVerbosity       :: !TracingVerbosity
+  , traceChainDB         :: !Bool
+    -- ^ By default we use 'readableChainDB' tracer, if on this it will use
+    -- more verbose tracer
+  , traceConsensus       :: ConsensusTraceOptions
+  , traceProtocols       :: ProtocolTraceOptions
+  , traceIpSubscription  :: !Bool
+  , traceDnsSubscription :: !Bool
+  , traceDnsResolver     :: !Bool
+  , traceMux             :: !Bool
+  } deriving (Eq, Show)
+
+deriving instance Show TracingVerbosity
+
+consensusTraceDefault :: ConsensusTraceOptions
+consensusTraceDefault =
+  ConsensusTracers.Tracers
+    (Const True)
+    (Const True)
+    (Const True)
+    (Const True)
+    (Const True)
+    (Const True)
+    (Const True)
+    (Const True)
+    (Const True)
+    (Const True)
+    (Const True)
+
+protocolTraceDefault :: ProtocolTraceOptions
+protocolTraceDefault =
+  ProtocolTracers
+    (Const True)
+    (Const True)
+    (Const True)
+    (Const True)
+    (Const True)
```

Comment:
Do we really want the default trace options trace _everything_?

My impression was that the defaults should be somewhat reasonable, or at least not overwhelming.. : -)

cc @CodiePP

Reply:
@deepfire Currently even when I don't list a `--log-config` option and I make the default values all `False` the logging output does not change. I will have to investigate this in a separate PR.